R/create_package.R

```diff
@@ -1,24 +1,46 @@
-#' Create an empty Data Package
+#' Create a Data Package
 #'
-#' Initiates a list describing a [Data
-#' Package](https://specs.frictionlessdata.io/data-package/).
-#' This empty Data Package can be extended with metadata and resources (see
-#' [add_resource()]).
-#' Added resources will make the Data Package meet [Tabular Data
-#' Package](https://specs.frictionlessdata.io/tabular-data-package/)
-#' requirements, so `profile` is set to `tabular-data-package`.
+#' Initiates a [Data Package](https://specs.frictionlessdata.io/data-package/)
+#' object, either from scratch or from an existing list.
+#' This Data Package object is a list with a `datapackage` class and the
+#' following properties:
+#' - All properties of the original `descriptor`.
+#' - [`resources`](
+#'   https://specs.frictionlessdata.io/data-package/#required-properties) (an
+#'   empty list) if not present.
+#' - `directory` (set to `"."` for the current directory) if not present.
+#'   It is used as the base path to access resources with [read_resource()].
 #'
-#' @return List describing a Data Package.
+#' The function will run [check_package()] on the created package to make sure
+#' it is valid.
+#'
+#' @param descriptor List to be made into a Data Package object.
+#'   If `NULL`, an empty Data Package object will be created from scratch.
+#' @return Data Package object.
 #' @family create functions
 #' @export
 #' @examples
 #' # Create a Data Package
 #' package <- create_package()
 #' str(package)
-create_package <- function() {
-  list(
-    profile = "tabular-data-package",
-    resources = list(),
-    directory = "." # Current directory
-  )
+create_package <- function(descriptor = NULL) {
+  if (!is.null(descriptor) && !is.list(descriptor)) {
+    cli::cli_abort(
+      "{.arg descriptor} must be a list if provided.",
+      class = "frictionless_error_descriptor_invalid"
+    )
+  }
+
+  # Add properties
+  descriptor$resources <- replace_null(descriptor$resources, list())
+  descriptor$directory <- replace_null(descriptor$directory, ".") # Current dir
+
+  # Add datapackage class
+  if (!"datapackage" %in% class(descriptor)) {
+    class(descriptor) <- c("datapackage", class(descriptor))
+  }
+
+  # Check that created package is valid
+  check_package(descriptor)
+  return(descriptor)
```

Comment:
```suggestion
  check_package(descriptor)
  
  return(descriptor)
```

I'd vertically split `check_package()` from `return()` because the comment above is not about `return()`